### PR TITLE
Basic level generation and rendering

### PR DESCRIPTION
**Pull request by raywan**

### CODE REVIEW

File: src/map.cpp [hunk 1]
```diff
@@ -22,6 +23,52 @@ Map::Map() : dim_x( MAP_LENGTH ), dim_y( MAP_LENGTH ) {
...
+  dim_x = map_lines[0].size();
+  curr_map = new int[ dim_x * dim_y ];
+
+  for (int i = 0; i < dim_y; i++) {
+    for (int j = 0; j < dim_x; j++) {
+      if (map_lines[i][j] == '.') {
+        set_tile(j, i, get_tile_index(6, 2));
+      } else if (map_lines[i][j] == 's') {
+        // TODO(ray): I don't really like how we have to consider the scale of rendering
+        // when we do positioning. Maybe finding a way to position before scaling everything.
```
Comment (zuqini) [inline src/map.cpp]: Agreed, we should have a global scaling factor that's applied when rendering, but not in our simulation.

File: scripts/ca.c [hunk 2]
```diff
@@ -0,0 +1,166 @@
+#include <stdio.h>
```
Comment (zuqini) [inline scripts/ca.c]: is this file just for prototyping?

File: src/map.cpp [hunk 3]
```diff
@@ -43,16 +90,16 @@ bool Map::is_solid( float x, float y) const {
 
 void Map::set_tile( int x, int y, int index ) {
   if ( x < 0 || x >= dim_x || y < 0 || y >= dim_y) return;
-  curr_map[ x * dim_y + y ] = index;
+  curr_map[ x * dim_x + y ] = index;
```
Comment (zuqini) [inline src/map.cpp]: This isn't right, it needs to be dim_y if the map isn't a square.

Comment (raywan) [reply src/map.cpp]: substitution mistake

**Pull request merged**